### DISCUSSION TITLE
Has anyone extended YUView to read files that have been gzipped?

### DISCUSSION THREAD

**Discussion by S-Fe** [IENT/YUView#561] [Ideas]
Hi all,
Firstly, thanks for producing such a great tool.

I was just curious if anyone has tried extending YUView to read files via zlib so that it can handle a gzipped file?  We have quite a few .yuv files and they occupy quite a bit of space.  A quick experiment suggests gzip might get a 50 to 75% saving in storage space.

I suppose the downside is that trying to implement random jumping to a particular frame may become 'less pleasant' but the storage and bandwidth savings for files on a network could be beneficial.

Simon

**Comment by ChristianFeldmann** [comment]
Hi! 

Good to hear that you like the tool. 

I have indeed seen zipped or gzipped yuv files a lot when I was working more in the mpeg standardization world. They are using this exactly for that reason. You can save >= 50% of bandwidth without any loss of information. 

However, I am unsure how well it would work to try to access the data in a compressed file. These formats are not meant for random access (unlike a video container format that is exactly designed for this). Maybe an alternative for you could be to put the video into a container and compress the video stream with the [FFV1 codec](https://trac.ffmpeg.org/wiki/Encode/FFV1) which is a lossless video codec. This may even give you better performance then gzip/zip. 

If it really has to be gzip/zip then it could be that the only way to make it work is to fully extract the full video ... maybe. But its a cool small coding idea. :)

Christian
